integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientExternalIdTest.java

```diff
@@ -0,0 +1,320 @@
+        final PostClientsResponse addClientResponse = ClientHelper.addClientAsPerson(requestSpec, responseSpec, jsonPayload);
+        final String clientExternalId = addClientResponse.getResourceExternalId();
+        final Long clientId = addClientResponse.getClientId();
+        assertNotNull(clientExternalId);
+        log.info("Client data id {} and external Id {}", clientId, clientExternalId);
+
+        GetClientsClientIdResponse clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.ACTIVE, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Close Client action
+        jsonPayload = clientHelper.getCloseClientAsJSON();
+        PostClientsClientIdResponse commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec,
+                clientExternalId, ClientHelper.CLOSE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        assertNotNull(commandResponse.getResourceExternalId());
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.CLOSED, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Reactivate Client action
+        jsonPayload = clientHelper.getReactivateClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REACTIVATE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        assertNotNull(commandResponse.getResourceExternalId());
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.PENDING, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Reject Client action
+        jsonPayload = clientHelper.getRejectClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REJECT_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        assertNotNull(commandResponse.getResourceExternalId());
```

Comment:
You know the value of the external id, so please do equality check instead of null check here

Reply:
Done